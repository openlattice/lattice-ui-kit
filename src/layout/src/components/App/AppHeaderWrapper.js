/*
 * @flow
 */

import React, { Children, Component } from 'react';
import type { Node } from 'react';

import isArray from 'lodash/isArray';
import isPlainObject from 'lodash/isPlainObject';
import styled from 'styled-components';
import {
  List,
  Map,
  get,
  isCollection,
} from 'immutable';

import * as Colors from '../../../../colors';
import { Button } from '../../../../button';
import { Select } from '../../../../select';
import { AppContentInnerWrapper, AppContentOuterWrapper } from './AppContentWrapper';
import { APP_CONTENT_PADDING } from '../../../../style/Sizes';

const { NEUTRALS, PURPLES, WHITE } = Colors;
const NAV_ROOT :string = 'nav-root';

const AppHeaderOuterWrapper = styled(AppContentOuterWrapper).attrs({
  as: 'header',
  bgColor: WHITE,
})`
  border-bottom: 1px solid ${NEUTRALS[5]};
`;

const AppHeaderInnerWrapper = styled(AppContentInnerWrapper)`
  align-items: center;
  flex-direction: row;
  justify-content: space-between;
  padding: 0 ${APP_CONTENT_PADDING}px;
`;

// button line-height = font-size (12px) * desired line-height (1.5) = 18px
// total button height = line-height (18px) + padding (2*8px) + border (2*1px) = 36px
const LogoutButton = styled(Button)`
  border: solid 1px ${NEUTRALS[4]};
  font-size: 12px;
  line-height: 18px;
  padding: 8px 16px;
  width: 100px;
`;

const HeaderRightContentWrapper = styled.div`
  align-items: center;
  display: flex;
  flex: 1 1 auto; /* this lets the select stretch a bit until max-width */
  font-size: 12px;
  font-weight: normal;
  justify-content: flex-end;
  padding: 12px 0;

  > * {
    margin-left: 30px;
  }
`;

const NavigationContentWrapper = styled.nav`
  display: flex;
  flex: 0 0 auto;
  height: 100%;
  min-height: 60px;
  justify-content: flex-start;

  > a {
    align-items: center;
    border-bottom: 3px solid transparent;
    color: ${NEUTRALS[1]};
    display: flex;
    font-size: 12px;
    font-weight: normal;
    justify-content: center;
    letter-spacing: normal;
    line-height: 18px; /* font-size (12px) * desired line-height (1.5) = 18px */
    margin: 0 0 0 30px;
    outline: none;
    padding: 3px 3px 0px 3px;
    text-align: center;
    text-decoration: none;

    &:first-child {
      margin: 0;
    }

    &:focus {
      outline: none;
      text-decoration: none;
    }

    &:hover {
      border-bottom: 3px solid ${NEUTRALS[2]};
      color: ${NEUTRALS[0]};
      cursor: pointer;
    }

    &.active {
      border-bottom: 3px solid ${PURPLES[1]};
      color: ${PURPLES[1]};
    }

    &.${NAV_ROOT} {
      border: none;
      margin: 0;
      padding: 0;

      > h1 {
        color: ${NEUTRALS[0]};
        font-size: 14px;
        font-weight: 600;
        letter-spacing: normal;
        margin: 0;
        padding: 0;
      }

      > img {
        height: 26px;
        margin-right: 10px;
      }
    }
  }
`;

const organizationsSelectStyles = {
  clearIndicator: (base) => ({ ...base, padding: '0', margin: '5px' }),
  container: (base, state) => ({
    ...base,
    cursor: state.isDisabled ? 'not-allowed' : 'default',
    pointerEvents: 'auto', // not sure what this was for
    // the following are not ideal, but it's tricky figuring out how to make it flex perfectly in all cases
    flex: '1 1 auto',
    maxWidth: '300px',
    minWidth: '200px',
  }),
  control: (base, state) => {

    const { isFocused, isDisabled, selectProps } = state;
    let backgroundColor = isFocused ? WHITE : NEUTRALS[8];
    let border = isFocused ? `solid 1px ${PURPLES[1]}` : `solid 1px ${NEUTRALS[4]}`;

    if (selectProps && selectProps.noBorder) {
      backgroundColor = 'transparent';
      border = 'none';
    }

    const style = {
      backgroundColor,
      border,
      borderRadius: '3px',
      boxShadow: 'none',
      fontSize: '12px',
      lineHeight: '18px', // 12px font size * 1.5 = 18px
      minHeight: '36px', // to match log out button total height
      pointerEvents: isDisabled ? 'none' : 'auto',
      ':hover': {
        backgroundColor,
        border,
      },
    };
    return { ...base, ...style };
  },
  dropdownIndicator: (base, state) => ({
    ...base,
    color: NEUTRALS[2],
    display: state.selectProps && state.selectProps.hideMenu ? 'none' : 'flex',
    margin: '4px',
    padding: '0',
  }),
  indicatorSeparator: () => ({ display: 'none' }),
  indicatorsContainer: (base) => ({ ...base, marginRight: '5px', color: NEUTRALS[2] }),
  menuList: (base) => ({ ...base, borderRadius: '4px' }),
  menuPortal: (base) => ({ ...base, zIndex: 1000 }),
  option: (base, state) => {

    const { isFocused, isSelected } = state;
    const color = isSelected ? PURPLES[1] : NEUTRALS[0];
    let backgroundColor = WHITE;
    if (isSelected) {
      backgroundColor = PURPLES[6];
    }
    else if (isFocused) {
      backgroundColor = NEUTRALS[6];
    }

    return {
      ...base,
      color,
      backgroundColor,
      fontSize: '12px',
      ':active': {
        backgroundColor: PURPLES[5],
      },
    };
  },
  singleValue: (base, state) => ({ ...base, color: state.isDisabled ? NEUTRALS[1] : NEUTRALS[0] }),
  valueContainer: (base) => ({ ...base, padding: '0 10px' }),
};

const AppIcon = ({ icon } :Object) => (
  icon ? <img alt="OpenLattice Application Icon" src={icon} /> : null
);

const AppTitle = ({ title = 'OpenLattice' } :Object) => (
  <h1>{title}</h1>
);

type Props = {
  children :Node;
  className :?string;
  icon :any;
  logout :() => void;
  organizationsSelect :{
    isDisabled :boolean;
    isLoading :boolean;
    onChange :(selectedOption :Object) => void;
    organizations :any;
    selectedOrganizationId :UUID;
  };
  user :?string;
};

type State = {
  shouldWrapNav :boolean;
};

class AppHeaderWrapper extends Component<Props, State> {

  headerRef :{ current :null | HTMLElement } = React.createRef();
  nav1Ref :{ current :null | HTMLElement } = React.createRef();
  nav2Ref :{ current :null | HTMLElement } = React.createRef();
  rightRef :{ current :null | HTMLElement } = React.createRef();

  constructor(props :Props) {

    super(props);
    this.state = {
      shouldWrapNav: false,
    };
  }

  componentDidMount() {

    // from the gods themselves... https://stackoverflow.com/a/19014495/196921
    window.addEventListener('resize', this.handleOnResize.bind(this));

    this.wrapNavIfNecessary();
  }

  componentDidUpdate() {

    this.wrapNavIfNecessary();
  }

  componentWillUnmount() {

    window.removeEventListener('resize', this.handleOnResize.bind(this));
  }

  handleOnResize = () => {

    const header = this.headerRef.current;
    const right = this.rightRef.current;
    const nav1 = this.nav1Ref.current;
    const nav2 = this.nav2Ref.current;

    // the lower nav ref will be null if it is not rendered, so this is to check if we should wrap the nav
    if (header && !nav2) {
      if (header.offsetWidth < header.scrollWidth) {
        this.setState({ shouldWrapNav: true });
      }
    }
    // check if the nav can fit in the header and undo the wrap if it can
    else if (header && nav1 && nav2 && right) {
      // this is a *super* basic estimate for checking if all header content will fit in a single row
      let headerWidthEstimate = 0;
      for (let i = 0; i < right.children.length; i += 1) {
        headerWidthEstimate += right.children[i].offsetWidth;
      }
      headerWidthEstimate += nav1.offsetWidth;
      headerWidthEstimate += nav2.offsetWidth;
      if (headerWidthEstimate < header.offsetWidth) {
        this.setState({ shouldWrapNav: false });
      }
    }
    // I don't think this is possible, but it's here for just in case
    else {
      this.setState({ shouldWrapNav: false });
    }
  }

  wrapNavIfNecessary = () => {

    // initially on mount, we know the lower nav will not render and the ref will be null
    const header = this.headerRef.current;
    if (header && header.offsetWidth < header.scrollWidth) {
      this.setState({ shouldWrapNav: true });
    }
  }

  renderHeaderRight = () => {

    const {
      logout,
      organizationsSelect = {},
      user,
    } = this.props;

    let organizations = [];
    if (isArray(organizationsSelect.organizations) || isPlainObject(organizationsSelect.organizations)) {
      organizations = Object.values(organizationsSelect.organizations);
    }
    else if (isCollection(organizationsSelect.organizations)) {
      organizations = organizationsSelect.organizations.valueSeq();
    }

    const organizationOptions = [];
    organizations.forEach((organization) => {
      organizationOptions.push({
        label: get(organization, 'title'),
        value: get(organization, 'id'),
      });
    });

    const selectedOrganizationOption = organizationOptions.find((option) => (
      option.value === organizationsSelect.selectedOrganizationId
    ));

    return (
      <HeaderRightContentWrapper ref={this.rightRef}>
        {
          user && (
            <span>{user}</span>
          )
        }
        {
          organizationOptions.length > 0 && (
            <Select
                isClearable={false}
                isDisabled={organizationsSelect.isDisabled}
                isLoading={organizationsSelect.isLoading}
                isMulti={false}
                menuPortalTarget={document.body}
                onChange={organizationsSelect.onChange}
                options={organizationOptions}
                placeholder="Select an organization..."
                styles={organizationsSelectStyles}
                value={selectedOrganizationOption} />
          )
        }
        <LogoutButton onClick={logout}>Log Out</LogoutButton>
      </HeaderRightContentWrapper>
    );
  }

  render() {

    const { children, className, icon } = this.props;
    const { shouldWrapNav } = this.state;

    return (
      <>
        <AppHeaderOuterWrapper className={className} ref={this.headerRef}>
          <AppHeaderInnerWrapper>
            <NavigationContentWrapper ref={this.nav1Ref}>
              {
                Children.map(children, (child, index) => {
                  if (index === 0) {
                    return React.cloneElement(
                      child,
                      { ...child.props, className: NAV_ROOT },
                      React.createElement(AppIcon, { icon }),
                      React.createElement(AppTitle, { title: child.props.children }),
                    );
                  }
                  return shouldWrapNav ? null : child;
                })
              }
            </NavigationContentWrapper>
            {this.renderHeaderRight()}
          </AppHeaderInnerWrapper>
        </AppHeaderOuterWrapper>
        {
          shouldWrapNav && (
            <AppHeaderOuterWrapper className={className}>
              <AppHeaderInnerWrapper>
                <NavigationContentWrapper ref={this.nav2Ref}>
                  {Children.map(children, (child, index) => (index !== 0 ? child : null))}
                </NavigationContentWrapper>
              </AppHeaderInnerWrapper>
            </AppHeaderOuterWrapper>
          )
        }
      </>
    );
  }
}

export default AppHeaderWrapper;
export {
  AppHeaderInnerWrapper,
  AppHeaderOuterWrapper,
};
