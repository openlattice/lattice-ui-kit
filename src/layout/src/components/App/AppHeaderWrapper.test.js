import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';

import AppHeaderWrapper from './AppHeaderWrapper';

const mockOrgs = {
  org1: { id: 'org1', title: 'OpenLattice' },
  org2: { id: 'org2', title: 'JustBeamIt' },
  org3: { id: 'org3', title: 'Metallica' },
};

const mockTabs = [
  <a key="home" href="/home">Home</a>,
  <a key="tab1" href="/tab1">Tab1</a>,
  <a key="tab2" href="/tab2">Tab2</a>,
];

const logout = () => {};

describe('AppHeaderWrapper', () => {

  describe('snapshots', () => {

    test('default', () => {
      const wrapper = shallow(
        <AppHeaderWrapper />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('logout', () => {
      const wrapper = shallow(
        <AppHeaderWrapper logout={logout} />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('navigation', () => {
      const wrapper = shallow(
        <AppHeaderWrapper logout={logout}>
          <div>{mockTabs}</div>
        </AppHeaderWrapper>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('navigation wrapping', () => {
      const wrapper = shallow(
        <AppHeaderWrapper logout={logout}>
          <div>{mockTabs}</div>
        </AppHeaderWrapper>
      );
      wrapper.setState({ shouldWrapNavigation: true });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('select', () => {
      const wrapper = shallow(
        <AppHeaderWrapper logout={logout} organizationsSelect={{ organizations: mockOrgs }} />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('select + navigation', () => {
      const wrapper = shallow(
        <AppHeaderWrapper logout={logout} organizationsSelect={{ organizations: mockOrgs }}>
          <div>{mockTabs}</div>
        </AppHeaderWrapper>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('user', () => {
      const wrapper = shallow(
        <AppHeaderWrapper logout={logout} user="jest@openlattice.com" />
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('user + navigation', () => {
      const wrapper = shallow(
        <AppHeaderWrapper logout={logout} user="jest@openlattice.com">
          <div>{mockTabs}</div>
        </AppHeaderWrapper>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('user + navigation + select', () => {
      const wrapper = shallow(
        <AppHeaderWrapper logout={logout} organizationsSelect={{ organizations: mockOrgs }} user="jest@openlattice.com">
          <div>{mockTabs}</div>
        </AppHeaderWrapper>
      );
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('force drawer', () => {
      const wrapper = shallow(
        <AppHeaderWrapper logout={logout} organizationsSelect={{ organizations: mockOrgs }} user="jest@openlattice.com">
          <div>{mockTabs}</div>
        </AppHeaderWrapper>
      );
      wrapper.setState({ shouldForceDrawer: true });
      expect(toJson(wrapper)).toMatchSnapshot();
    });

  });

});
