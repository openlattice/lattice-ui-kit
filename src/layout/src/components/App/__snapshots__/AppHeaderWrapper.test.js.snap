// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeaderWrapper snapshots default 1`] = `
<Fragment>
  <AppHeaderOuterWrapper
    onClick={[Function]}
  >
    <AppHeaderInnerWrapper>
      <NavigationWrapper>
        <a
          className="app-nav-root"
          href="http://localhost/"
        >
          <AppIcon />
          <AppTitle />
        </a>
      </NavigationWrapper>
      <AppHeaderWrapper__HeaderSectionContentWrapper
        align="right"
      />
    </AppHeaderInnerWrapper>
  </AppHeaderOuterWrapper>
</Fragment>
`;

exports[`AppHeaderWrapper snapshots force drawer 1`] = `
<Fragment>
  <AppHeaderOuterWrapper
    onClick={[Function]}
  >
    <AppHeaderInnerWrapper>
      <NavigationWrapper
        key=".0"
      >
        <a
          className="app-nav-root"
          href="/home"
          key=".$home"
        >
          <AppIcon />
        </a>
      </NavigationWrapper>
      <AppHeaderWrapper__HeaderSectionContentWrapper
        align="right"
      >
        <Select
          isClearable={false}
          isMulti={false}
          menuPortalTarget={<body />}
          options={
            Array [
              Object {
                "label": "OpenLattice",
                "value": "org1",
              },
              Object {
                "label": "JustBeamIt",
                "value": "org2",
              },
              Object {
                "label": "Metallica",
                "value": "org3",
              },
            ]
          }
          placeholder="Select an organization..."
          styles={
            Object {
              "clearIndicator": [Function],
              "container": [Function],
              "control": [Function],
              "dropdownIndicator": [Function],
              "indicatorSeparator": [Function],
              "indicatorsContainer": [Function],
              "menu": [Function],
              "menuList": [Function],
              "menuPortal": [Function],
              "option": [Function],
              "singleValue": [Function],
              "valueContainer": [Function],
            }
          }
        />
        <AppHeaderWrapper__NavigationToggleWrapper
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className=""
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f0c9",
                  "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                ],
                "iconName": "bars",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </AppHeaderWrapper__NavigationToggleWrapper>
      </AppHeaderWrapper__HeaderSectionContentWrapper>
    </AppHeaderInnerWrapper>
  </AppHeaderOuterWrapper>
  <ForwardRef
    drawer={true}
    isOpen={false}
    key=".0"
    onClose={[Function]}
  >
    <span>
      jest@openlattice.com
    </span>
    <hr />
    <a
      href="/tab1"
      key=".$tab1"
      onClick={[Function]}
    >
      Tab1
    </a>
    <a
      href="/tab2"
      key=".$tab2"
      onClick={[Function]}
    >
      Tab2
    </a>
    <AppHeaderWrapper__LogOutAnchor
      onClick={[Function]}
    >
      <FontAwesomeIcon
        border={false}
        className=""
        fixedWidth={true}
        flip={null}
        icon={
          Object {
            "icon": Array [
              512,
              512,
              Array [],
              "f08b",
              "M96 64h84c6.6 0 12 5.4 12 12v24c0 6.6-5.4 12-12 12H96c-26.5 0-48 21.5-48 48v192c0 26.5 21.5 48 48 48h84c6.6 0 12 5.4 12 12v24c0 6.6-5.4 12-12 12H96c-53 0-96-43-96-96V160c0-53 43-96 96-96zm231.1 19.5l-19.6 19.6c-4.8 4.8-4.7 12.5.2 17.1L420.8 230H172c-6.6 0-12 5.4-12 12v28c0 6.6 5.4 12 12 12h248.8L307.7 391.7c-4.8 4.7-4.9 12.4-.2 17.1l19.6 19.6c4.7 4.7 12.3 4.7 17 0l164.4-164c4.7-4.7 4.7-12.3 0-17l-164.4-164c-4.7-4.6-12.3-4.6-17 .1z",
            ],
            "iconName": "sign-out",
            "prefix": "far",
          }
        }
        inverse={false}
        listItem={false}
        mask={null}
        pull={null}
        pulse={false}
        rotation={null}
        size="lg"
        spin={false}
        swapOpacity={false}
        symbol={false}
        title=""
        transform={null}
      />
      <span
        style={
          Object {
            "marginLeft": "20px",
          }
        }
      >
        Log Out
      </span>
    </AppHeaderWrapper__LogOutAnchor>
  </ForwardRef>
</Fragment>
`;

exports[`AppHeaderWrapper snapshots logout 1`] = `
<Fragment>
  <AppHeaderOuterWrapper
    onClick={[Function]}
  >
    <AppHeaderInnerWrapper>
      <NavigationWrapper>
        <a
          className="app-nav-root"
          href="http://localhost/"
        >
          <AppIcon />
          <AppTitle />
        </a>
      </NavigationWrapper>
      <AppHeaderWrapper__HeaderSectionContentWrapper
        align="right"
      >
        <AppHeaderWrapper__LogoutButton
          onClick={[Function]}
        >
          Log Out
        </AppHeaderWrapper__LogoutButton>
      </AppHeaderWrapper__HeaderSectionContentWrapper>
    </AppHeaderInnerWrapper>
  </AppHeaderOuterWrapper>
</Fragment>
`;

exports[`AppHeaderWrapper snapshots navigation 1`] = `
<Fragment>
  <AppHeaderOuterWrapper
    onClick={[Function]}
  >
    <AppHeaderInnerWrapper>
      <NavigationWrapper
        key=".0"
      >
        <a
          className="app-nav-root"
          href="/home"
          key=".$home"
        >
          <AppIcon />
          <AppTitle
            title="Home"
          />
        </a>
        <a
          href="/tab1"
          key=".$tab1"
        >
          Tab1
        </a>
        <a
          href="/tab2"
          key=".$tab2"
        >
          Tab2
        </a>
      </NavigationWrapper>
      <AppHeaderWrapper__HeaderSectionContentWrapper
        align="right"
      >
        <AppHeaderWrapper__LogoutButton
          onClick={[Function]}
        >
          Log Out
        </AppHeaderWrapper__LogoutButton>
      </AppHeaderWrapper__HeaderSectionContentWrapper>
    </AppHeaderInnerWrapper>
  </AppHeaderOuterWrapper>
</Fragment>
`;

exports[`AppHeaderWrapper snapshots navigation wrapping 1`] = `
<Fragment>
  <AppHeaderOuterWrapper
    onClick={[Function]}
  >
    <AppHeaderInnerWrapper>
      <NavigationWrapper
        key=".0"
      >
        <a
          className="app-nav-root"
          href="/home"
          key=".$home"
        >
          <AppIcon />
          <AppTitle
            title="Home"
          />
        </a>
      </NavigationWrapper>
      <AppHeaderWrapper__HeaderSectionContentWrapper
        align="right"
      >
        <AppHeaderWrapper__LogoutButton
          onClick={[Function]}
        >
          Log Out
        </AppHeaderWrapper__LogoutButton>
      </AppHeaderWrapper__HeaderSectionContentWrapper>
    </AppHeaderInnerWrapper>
  </AppHeaderOuterWrapper>
  <ForwardRef>
    <a
      href="/tab1"
      key=".0/.$.$tab1"
    >
      Tab1
    </a>
    <a
      href="/tab2"
      key=".0/.$.$tab2"
    >
      Tab2
    </a>
  </ForwardRef>
</Fragment>
`;

exports[`AppHeaderWrapper snapshots select + navigation 1`] = `
<Fragment>
  <AppHeaderOuterWrapper
    onClick={[Function]}
  >
    <AppHeaderInnerWrapper>
      <NavigationWrapper
        key=".0"
      >
        <a
          className="app-nav-root"
          href="/home"
          key=".$home"
        >
          <AppIcon />
          <AppTitle
            title="Home"
          />
        </a>
        <a
          href="/tab1"
          key=".$tab1"
        >
          Tab1
        </a>
        <a
          href="/tab2"
          key=".$tab2"
        >
          Tab2
        </a>
      </NavigationWrapper>
      <AppHeaderWrapper__HeaderSectionContentWrapper
        align="right"
      >
        <Select
          isClearable={false}
          isMulti={false}
          menuPortalTarget={<body />}
          options={
            Array [
              Object {
                "label": "OpenLattice",
                "value": "org1",
              },
              Object {
                "label": "JustBeamIt",
                "value": "org2",
              },
              Object {
                "label": "Metallica",
                "value": "org3",
              },
            ]
          }
          placeholder="Select an organization..."
          styles={
            Object {
              "clearIndicator": [Function],
              "container": [Function],
              "control": [Function],
              "dropdownIndicator": [Function],
              "indicatorSeparator": [Function],
              "indicatorsContainer": [Function],
              "menu": [Function],
              "menuList": [Function],
              "menuPortal": [Function],
              "option": [Function],
              "singleValue": [Function],
              "valueContainer": [Function],
            }
          }
        />
        <AppHeaderWrapper__LogoutButton
          onClick={[Function]}
        >
          Log Out
        </AppHeaderWrapper__LogoutButton>
      </AppHeaderWrapper__HeaderSectionContentWrapper>
    </AppHeaderInnerWrapper>
  </AppHeaderOuterWrapper>
</Fragment>
`;

exports[`AppHeaderWrapper snapshots select 1`] = `
<Fragment>
  <AppHeaderOuterWrapper
    onClick={[Function]}
  >
    <AppHeaderInnerWrapper>
      <NavigationWrapper>
        <a
          className="app-nav-root"
          href="http://localhost/"
        >
          <AppIcon />
          <AppTitle />
        </a>
      </NavigationWrapper>
      <AppHeaderWrapper__HeaderSectionContentWrapper
        align="right"
      >
        <Select
          isClearable={false}
          isMulti={false}
          menuPortalTarget={<body />}
          options={
            Array [
              Object {
                "label": "OpenLattice",
                "value": "org1",
              },
              Object {
                "label": "JustBeamIt",
                "value": "org2",
              },
              Object {
                "label": "Metallica",
                "value": "org3",
              },
            ]
          }
          placeholder="Select an organization..."
          styles={
            Object {
              "clearIndicator": [Function],
              "container": [Function],
              "control": [Function],
              "dropdownIndicator": [Function],
              "indicatorSeparator": [Function],
              "indicatorsContainer": [Function],
              "menu": [Function],
              "menuList": [Function],
              "menuPortal": [Function],
              "option": [Function],
              "singleValue": [Function],
              "valueContainer": [Function],
            }
          }
        />
        <AppHeaderWrapper__LogoutButton
          onClick={[Function]}
        >
          Log Out
        </AppHeaderWrapper__LogoutButton>
      </AppHeaderWrapper__HeaderSectionContentWrapper>
    </AppHeaderInnerWrapper>
  </AppHeaderOuterWrapper>
</Fragment>
`;

exports[`AppHeaderWrapper snapshots user + navigation + select 1`] = `
<Fragment>
  <AppHeaderOuterWrapper
    onClick={[Function]}
  >
    <AppHeaderInnerWrapper>
      <NavigationWrapper
        key=".0"
      >
        <a
          className="app-nav-root"
          href="/home"
          key=".$home"
        >
          <AppIcon />
          <AppTitle
            title="Home"
          />
        </a>
        <a
          href="/tab1"
          key=".$tab1"
        >
          Tab1
        </a>
        <a
          href="/tab2"
          key=".$tab2"
        >
          Tab2
        </a>
      </NavigationWrapper>
      <AppHeaderWrapper__HeaderSectionContentWrapper
        align="right"
      >
        <span>
          jest@openlattice.com
        </span>
        <Select
          isClearable={false}
          isMulti={false}
          menuPortalTarget={<body />}
          options={
            Array [
              Object {
                "label": "OpenLattice",
                "value": "org1",
              },
              Object {
                "label": "JustBeamIt",
                "value": "org2",
              },
              Object {
                "label": "Metallica",
                "value": "org3",
              },
            ]
          }
          placeholder="Select an organization..."
          styles={
            Object {
              "clearIndicator": [Function],
              "container": [Function],
              "control": [Function],
              "dropdownIndicator": [Function],
              "indicatorSeparator": [Function],
              "indicatorsContainer": [Function],
              "menu": [Function],
              "menuList": [Function],
              "menuPortal": [Function],
              "option": [Function],
              "singleValue": [Function],
              "valueContainer": [Function],
            }
          }
        />
        <AppHeaderWrapper__LogoutButton
          onClick={[Function]}
        >
          Log Out
        </AppHeaderWrapper__LogoutButton>
      </AppHeaderWrapper__HeaderSectionContentWrapper>
    </AppHeaderInnerWrapper>
  </AppHeaderOuterWrapper>
</Fragment>
`;

exports[`AppHeaderWrapper snapshots user + navigation 1`] = `
<Fragment>
  <AppHeaderOuterWrapper
    onClick={[Function]}
  >
    <AppHeaderInnerWrapper>
      <NavigationWrapper
        key=".0"
      >
        <a
          className="app-nav-root"
          href="/home"
          key=".$home"
        >
          <AppIcon />
          <AppTitle
            title="Home"
          />
        </a>
        <a
          href="/tab1"
          key=".$tab1"
        >
          Tab1
        </a>
        <a
          href="/tab2"
          key=".$tab2"
        >
          Tab2
        </a>
      </NavigationWrapper>
      <AppHeaderWrapper__HeaderSectionContentWrapper
        align="right"
      >
        <span>
          jest@openlattice.com
        </span>
        <AppHeaderWrapper__LogoutButton
          onClick={[Function]}
        >
          Log Out
        </AppHeaderWrapper__LogoutButton>
      </AppHeaderWrapper__HeaderSectionContentWrapper>
    </AppHeaderInnerWrapper>
  </AppHeaderOuterWrapper>
</Fragment>
`;

exports[`AppHeaderWrapper snapshots user 1`] = `
<Fragment>
  <AppHeaderOuterWrapper
    onClick={[Function]}
  >
    <AppHeaderInnerWrapper>
      <NavigationWrapper>
        <a
          className="app-nav-root"
          href="http://localhost/"
        >
          <AppIcon />
          <AppTitle />
        </a>
      </NavigationWrapper>
      <AppHeaderWrapper__HeaderSectionContentWrapper
        align="right"
      >
        <span>
          jest@openlattice.com
        </span>
        <AppHeaderWrapper__LogoutButton
          onClick={[Function]}
        >
          Log Out
        </AppHeaderWrapper__LogoutButton>
      </AppHeaderWrapper__HeaderSectionContentWrapper>
    </AppHeaderInnerWrapper>
  </AppHeaderOuterWrapper>
</Fragment>
`;
